asapdiscovery-docking/asapdiscovery/docking/workflows/large_scale_docking.py

```diff
@@ -0,0 +1,359 @@
+)
+from asapdiscovery.data.postera.postera_factory import PosteraFactory
+from asapdiscovery.data.postera.postera_uploader import PosteraUploader
+from asapdiscovery.data.schema_v2.fragalysis import FragalysisFactory
+from asapdiscovery.data.schema_v2.ligand import write_ligands_to_multi_sdf
+from asapdiscovery.data.schema_v2.molfile import MolFileFactory
+from asapdiscovery.data.schema_v2.structure_dir import StructureDirFactory
+from asapdiscovery.data.selectors.mcs_selector import MCSSelector
+from asapdiscovery.data.services_config import PosteraSettings
+from asapdiscovery.data.utils import check_empty_dataframe
+from asapdiscovery.docking.docking_data_validation import (
+    DockingResultColsV2 as DockingResultCols,
+)
+from asapdiscovery.docking.docking_v2 import POSITDocker
+from asapdiscovery.docking.scorer_v2 import (
+    ChemGauss4Scorer,
+    GATScorer,
+    MetaScorer,
+    SchnetScorer,
+)
+from asapdiscovery.modeling.protein_prep_v2 import ProteinPrepper
+from pydantic import (
+    BaseModel,
+    Field,
+    PositiveFloat,
+    PositiveInt,
+    root_validator,
+    validator,
+)
+
+
+class LargeScaleDockingInputs(BaseModel):
+    filename: Optional[str] = Field(
+        None, description="Path to a molecule file containing query ligands."
+    )
+    fragalysis_dir: Optional[str] = Field(
+        None, description="Path to a directory containing a Fragalysis dump."
+    )
+    structure_dir: Optional[str] = Field(
+        None,
+        description="Path to a directory containing structures to dock instead of a full fragalysis database.",
+    )
+    postera: bool = Field(
+        False, description="Whether to use the Postera database as the query set."
+    )
+    postera_upload: bool = Field(
+        False, description="Whether to upload the results to Postera."
+    )
+    postera_molset_name: Optional[str] = Field(
+        None, description="The name of the molecule set to upload to."
+    )
+    du_cache: Optional[str] = Field(
+        None, description="Path to a directory where design units are cached"
+    )
+
+    gen_du_cache: Optional[str] = Field(
+        None,
+        description="Path to a directory where generated design units should be cached",
+    )
+
+    target: TargetTags = Field(None, description="The target to dock against.")
+    write_final_sdf: bool = Field(
+        default=True,
+        description="Whether to write the final docked poses to an SDF file.",
+    )
+    use_dask: bool = Field(True, description="Whether to use dask for parallelism.")
+
+    dask_type: DaskType = Field(
+        DaskType.LOCAL, description="Dask client to use for parallelism."
+    )
+
+    n_select: PositiveInt = Field(
+        10, description="Number of targets to dock each ligand against, sorted by MCS"
+    )
+
+    top_n: PositiveInt = Field(
+        500, description="Number of docking results to return, ordered by docking score"
+    )
+
+    posit_confidence_cutoff: PositiveFloat = Field(
```

Comment:
We can use the less-than validator built into field here https://docs.pydantic.dev/1.10/usage/schema/#field-customization